tests/src/Kernel/AccessTest.php

```diff
@@ -0,0 +1,74 @@
+<?php
+
+declare(strict_types = 1);
+
+namespace Drupal\Tests\oe_contact_forms\Kernel;
+
+use Drupal\contact\Entity\ContactForm;
+use Drupal\Core\Access\AccessResult;
+
+/**
+ * Test the contact form access control.
+ */
+class AccessTest extends ContactFormTestBase {
+
+  /**
+   * Tests corporate contact form access.
+   */
+  public function testCorporateFormAccess(): void {
+    $contact_form = ContactForm::create(['id' => 'oe_contact_form']);
+    $contact_form->setThirdPartySetting('oe_contact_forms', 'is_corporate_form', TRUE);
+    $contact_form->save();
+
+    // Assert user with "access corporate contact form" permission has access
+    // to contact_form entity.
+    $account = $this->createUser(['name' => 'test1'], ['access corporate contact form']);
+    \Drupal::currentUser()->setAccount($account);
+    $this->assertTrue($contact_form->access('view', \Drupal::currentUser()));
+    // Assert user without "access corporate contact form" permission does not.
+    $account = $this->createUser(['name' => 'test2'], ['access site-wide contact form']);
+    \Drupal::currentUser()->setAccount($account);
+    $this->assertFalse($contact_form->access('view', \Drupal::currentUser()));
+
+    // Test access check service.
+    $view_access = \Drupal::service('oe_contact_forms.access_checker');
```

Comment:
No, don't test directly the access checker. Test the canonical route to the form is accessed correctly. This will include your access checker + together with the existing access handling on it.

Reply:
ok, I used PathValidator

Reply:
See other comment: why? :)